lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1454,6 +1454,22 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     Value mult = b.create<arith::MulFOp>(loc, self, alpha);
     return b.create<arith::SubFOp>(loc, other, mult);
   }
+  if (isa<AtenToDtypeOp>(op)) {
+    Value input = payloadArgs[0];
+    Type inType = input.getType();
+    Type outType = payloadArgs[1].getType();
+    Value result;
+    if (inType.isF32()) {
```

Comment:
The code need to be guarded make it clear we only support floating point for now.  You could refer to other ops in this function. https://github.com/llvm/torch-mlir/blob/127c7d8e27ede211af77392b5277ba06fb515fb7/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L1307-L1312.